### ISSUE TITLE
ScriptCreator.cs file missing

### ISSUE THREAD

**Issue by sandlogic** [MuhammadNoumanZafar/TestScriptGeneration#1]
Hello Muhammad

ScriptCreator.cs file is missing, can you please share the same?

Cheers
Kamal

**Comment by MuhammadNoumanZafar** [comment]
Hello!
Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information publicly.

**Comment by MuhammadNoumanZafar** [comment]
> Hello Muhammad
> 
> ScriptCreator.cs file is missing, can you please share the same?
> 
> Cheers
> Kamal

Hello!
Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information.

**Comment by sandlogic** [comment]
Understood. Can you outline the purpose of the file so that we try to mimic.CheersKamalSent from my Galaxy
-------- Original message --------From: Muhammad Nouman Zafar <notifications@github.com> Date: 06/01/2021  19:16  (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration <TestScriptGeneration@noreply.github.com> Cc: SandLogic Technologies <kamal@sandlogic.com>, Author <author@noreply.github.com> Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) 

Hello Muhammad
ScriptCreator.cs file is missing, can you please share the same?
Cheers
Kamal

Hello!
Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information.

**Comment by MuhammadNoumanZafar** [comment]
> Understood. Can you outline the purpose of the file so that we try to mimic.CheersKamalSent from my Galaxy -------- Original message --------From: Muhammad Nouman Zafar <notifications@github.com> Date: 06/01/2021 19:16 (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration <TestScriptGeneration@noreply.github.com> Cc: SandLogic Technologies <kamal@sandlogic.com>, Author <author@noreply.github.com> Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) Hello Muhammad ScriptCreator.cs file is missing, can you please share the same? Cheers Kamal Hello! Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information. —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

Yes, I can but if you can tell me why you need this information then I think I can be more specific in your context.

**Comment by sandlogic** [comment]
We are looking at ref implementation of Test Case generation given a requirement.Sent from my Galaxy
-------- Original message --------From: Muhammad Nouman Zafar <notifications@github.com> Date: 06/01/2021  19:41  (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration <TestScriptGeneration@noreply.github.com> Cc: SandLogic Technologies <kamal@sandlogic.com>, Author <author@noreply.github.com> Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) 



Understood. Can you outline the purpose of the file so that we try to mimic.CheersKamalSent from my Galaxy -------- Original message --------From: Muhammad Nouman Zafar notifications@github.com Date: 06/01/2021 19:16 (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration TestScriptGeneration@noreply.github.com Cc: SandLogic Technologies kamal@sandlogic.com, Author author@noreply.github.com Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) Hello Muhammad ScriptCreator.cs file is missing, can you please share the same? Cheers Kamal Hello! Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information. —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.



Yes, I can but if you can tell me why you need this information that I think I can be more specific in your context.

**Comment by MuhammadNoumanZafar** [comment]
> We are looking at ref implementation of Test Case generation given a requirement.Sent from my Galaxy -------- Original message --------From: Muhammad Nouman Zafar <notifications@github.com> Date: 06/01/2021 19:41 (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration <TestScriptGeneration@noreply.github.com> Cc: SandLogic Technologies <kamal@sandlogic.com>, Author <author@noreply.github.com> Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) Understood. Can you outline the purpose of the file so that we try to mimic.CheersKamalSent from my Galaxy -------- Original message --------From: Muhammad Nouman Zafar notifications@github.com Date: 06/01/2021 19:16 (GMT+05:30) To: MuhammadNoumanZafar/TestScriptGeneration TestScriptGeneration@noreply.github.com Cc: SandLogic Technologies kamal@sandlogic.com, Author author@noreply.github.com Subject: Re: [MuhammadNoumanZafar/TestScriptGeneration] ScriptCreator.cs file missing (#1) Hello Muhammad ScriptCreator.cs file is missing, can you please share the same? Cheers Kamal Hello! Actually, this prototype has been developed for CPS in collaboration with Bombardier Transportation AB, Sweden and the ScriptCreator.cs file contains some sensitive information related to SUT (i.e. information about custom build libraries). So, I am afraid I can not share such information. —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe. Yes, I can but if you can tell me why you need this information that I think I can be more specific in your context. —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

Okay! In that case, this prototype is basically based on our proposed framework (TIGER) which uses the GraphWalker (GW) for generating test cases. As GW is a Model based tool so the generated test cases are more at the abstract level, so our tool concertize the generated test cases (as you can see in the code too) and then generate test scripts using ScriptCreator.cs. As CPS deals with multiple signals so concretization process deals with concretization of signals and use the generated test script to make them executable. So, Let say hypothetically if you are using a robot framework to test a web application the ScriptCreator.cs will contain the information related to robort framework i.e. syntax and libraries etc. and will generate the files containing concrete test cases in similar format that can be directly executable on targeted SUT.
